identity-apps-core/apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -129,57 +175,168 @@
     }
 
     if (isUsernameRecovery) {
-        // Username Recovery Scenario
+        // Username recovery scenario.
+        String recoveryStage = IdentityManagementEndpointUtil.getStringValue(request.getAttribute("recoveryStage"));
+        if(recoveryStage == "" || recoveryStage == null){
+            recoveryStage = request.getParameter("recoveryStage");
+        }
+
+
         if (StringUtils.isBlank(tenantDomain)) {
             tenantDomain = IdentityManagementEndpointConstants.SUPER_TENANT;
         }
 
-        List<Claim> claims;
-        UsernameRecoveryApi usernameRecoveryApi = new UsernameRecoveryApi();
-        try {
-            claims = usernameRecoveryApi.getClaimsForUsernameRecovery(tenantDomain, true);
-        } catch (ApiException e) {
-            IdentityManagementEndpointUtil.addErrorInformation(request, e);
-            if (!StringUtils.isBlank(username)) {
-                request.setAttribute("username", username);
+        if(UsernameRecoveryStage.INITIATE.equalsValue(recoveryStage)){
+            
+            // Separate the contact to mobile or email.
+            String contact = request.getParameter("contact");
+            String mobileClaimRegex = "^\\s*(?:\\+?(\\d{1,3}))?[\\-. (]*(\\d{2,3})[\\-. )]*(\\d{3})[\\-. ]*(\\d{4,6})(?: *x(\\d+))?\\s*$";
```

Comment:
Shall we define this as a constant?

Reply:
Fixed by https://github.com/wso2/identity-apps/pull/7005/commits/f356f29765969368e431d4be7500caa7440ba6db